custom/icds_reports/static/js/angular-services/baseControllers.service.js

```diff
@@ -54,6 +54,16 @@ window.angular.module('icdsApp').factory('baseControllersService', ['$timeout',
             $scope.$emit('filtersChange');
         });
         vm.selectedMonthDisplay = dateHelperService.getSelectedMonthDisplay();
+        vm.selectedLocation = function () {
+            return storageService.getKey('selectedLocation');
+        };
+        vm.showReassignmentMessage = function () {
+            var selectedMonth = parseInt($location.search()['month']) || new Date().getMonth() + 1;
+            var selectedYear =  parseInt($location.search()['year']) || new Date().getFullYear();
+            var selectedDate = selectedMonth ? new Date(selectedYear, selectedMonth) : new Date();
+            var selectedLocation = vm.selectedLocation();
+            return selectedLocation && (Date.parse(selectedLocation.deprecated_at) < selectedDate || Date.parse(selectedLocation.deprecates_at) > selectedDate);
```

Comment:
Thinking from how I imagined this to be visible on dashboard, should not there be two different methods (showDeprecatedMessage & showDeprecatesMessage) to show message for a 
1. deprecated location
2. a location that deprecated another location

I was thinking of two different messages like

on a deprecated location we see
```
this location was extracted from ABC or
this location was split from ABC or
this location was a result of merge between ABC and DEF or
this location was moved from ABC
```

and

on the other location we see similar or simple like
```
this location deprecated DEF via merge/split/extract/move
```

Reply:
I think I am going to leave most of that logic for the angular template. `reassignment-message.html` can figure out how to render the message since it has the selected location, and this keeps the logic in the main controllers and the rest of the templates much simpler.